packages/mdc-dialog/index.js

```diff
@@ -77,17 +98,25 @@ export class MDCDialog extends MDCComponent {
       addBodyClass: (className) => document.body.classList.add(className),
       removeBodyClass: (className) => document.body.classList.remove(className),
       eventTargetHasClass: (target, className) => target.classList.contains(className),
+      eventTargetMatchesSelector: (target, selector) => matches(target, selector),
       registerInteractionHandler: (evt, handler) => this.root_.addEventListener(evt, handler),
       deregisterInteractionHandler: (evt, handler) => this.root_.removeEventListener(evt, handler),
-      registerSurfaceInteractionHandler: (evt, handler) => this.dialogSurface_.addEventListener(evt, handler),
-      deregisterSurfaceInteractionHandler: (evt, handler) => this.dialogSurface_.removeEventListener(evt, handler),
+      registerSurfaceInteractionHandler: (evt, handler) => this.containerElement_.addEventListener(evt, handler),
+      deregisterSurfaceInteractionHandler: (evt, handler) => this.containerElement_.removeEventListener(evt, handler),
       registerDocumentKeydownHandler: (handler) => document.addEventListener('keydown', handler),
       deregisterDocumentKeydownHandler: (handler) => document.removeEventListener('keydown', handler),
-      notifyAccept: () => this.emit(MDCDialogFoundation.strings.ACCEPT_EVENT),
+      isScrollable: (element) => element.scrollHeight > element.offsetHeight,
+      getContentElement: () => this.contentElement_,
+      getButtonElements: () => this.buttonElements_,
+      notifyYes: () => this.emit(MDCDialogFoundation.strings.YES_EVENT),
```

Comment:
Rather than hard-code to yes/no/cancel (which are likely to confuse people, because sometimes a button with a "no" label is actually a "cancel" button i.e. no action), we should document a data attribute to apply to action buttons (and _not_ apply to cancel), which we can pick up and report via a property on `event.detail` in the closing/closed events.

In order to do this, since `closed` in particular is asynchronous, we will need to cache the value internally... if we really want to be safe from race conditions (i.e. a user hitting 2 buttons really fast before the transition ends), we should:

1. Clear the internal property on open
2. When an action button is clicked, check whether the property has a value
3. If the property doesn't yet have a value when clicked, update it